.github/workflows/deploy.yaml

```diff
@@ -0,0 +1,49 @@
+# Adapted from sample vite workflow file: https://vitejs.dev/guide/static-deploy#github-pages
+name: deploy
+
+on:
+  release:
+    types: [published]
+  workflow_dispatch:
+
+# Sets the GITHUB_TOKEN permissions to allow deployment to GitHub Pages
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+
+# Allow one concurrent deployment
+concurrency:
+  group: 'pages'
+  cancel-in-progress: true
+
+jobs:
+  deploy:
+    environment:
+      name: github-pages
+      url: ${{ steps.deployment.outputs.page_url }}
```

Comment:
I know this is from a template file, but what is going on here? This is an environment variable for the **job**, that is defined based on the output of the last **step** in this **job**? Is this environment variable used by any of the other steps in the job? And if so: how does it have any value before the last step is run?

Reply:
OK, answering my own question here: **environment** is a special word from GitHub for **deployments**. And it has attributes that can be customized like 
- name
- url

So what is happening here is that we define a GitHub deployment environment based on the output of the last step in this deployment job. And what will happen is that after the deployment, this environment will show up on the GitHub UI as one of the deployment environments: https://docs.github.com/en/actions/deployment/targeting-different-environments/using-environments-for-deployment#using-an-environment

Maybe worth linking to the GitHub docs here so we remember this?

Reply:
I.e. here it shows up: 
![image](https://github.com/neurobagel/react-query-tool/assets/1302022/3b3e4d0e-dfb2-43f8-88c8-c1c387d67785)